presidio-analyzer/presidio_analyzer/predefined_recognizers/es_nie_recognizer.py

```diff
@@ -0,0 +1,67 @@
+from typing import List, Tuple, Optional
+
+from presidio_analyzer import Pattern, PatternRecognizer
+
+
+class EsNieRecognizer(PatternRecognizer):
+    """
+    Recognize NIE number using regex and checksum.
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+    for different strings to be used during pattern matching.
+    This can allow a greater variety in input, for example by removing dashes or spaces.  
```

Comment:
Can you please add a link to the source of the regex and checksum logic as a comment?

Reply:
Of course:

- https://es.wikipedia.org/wiki/N%C3%BAmero_de_identidad_de_extranjero
- https://www.interior.gob.es/opencms/ca/servicios-al-ciudadano/tramites-y-gestiones/dni/calculo-del-digito-de-control-del-nif-nie/
- https://www.juntadeandalucia.es/servicios/madeja/contenido/libro-pautas/196
- https://www.juntadeandalucia.es/servicios/madeja/contenido/recurso/677
- https://www.policia.es/_es/extranjeria_extranjeros.php
- https://www.boe.es/buscar/act.php?id=BOE-A-1997-3364
- https://www.boe.es/buscar/doc.php?id=BOE-A-2008-12050
-

Reply:
Thanks! Please add one of those (say the second link) in a code comment